src/execution/mod.rs

```diff
@@ -129,47 +129,44 @@ where
         let mut wasm = WasmReader::new(self.wasm_bytecode);
         wasm.move_start_to(inst.code_expr);
 
+        use crate::core::reader::types::opcode::*;
         loop {
             // call the instruction hook
             #[cfg(feature = "hooks")]
             H::instruction_hook(self);
 
-            match wasm.read_u8().unwrap_validated() {
-                // end
-                0x0B => {
+            let first_byte = wasm.read_u8().unwrap_validated();
```

Comment:
I see the value in making a variable with the first byte, but maybe also rename it to something more suggestive: first_instr_byte